ejb3/src/main/java/org/jboss/as/ejb3/timerservice/TimerServiceImpl.java

```diff
@@ -1133,12 +1128,23 @@ public TimerServiceResource getResource() {
         return resource;
     }
 
-    private void registerTimerResource(final String timerId) {
-        this.resource.timerCreated(timerId);
+    private boolean registerTimerResource(final TimerImpl timer) {
+        synchronized (this.timers) {
+            if (this.timers.containsKey(timer.getId())) {
```

Comment:
`timers` is already declared as a synchronized map. We may want to switch to a regular `HashMap` and guard all access to it to avoid double synchronize it.

Reply:
done